### PR TITLE
Fix: Chart date fix, add functions to make correct request to CG API

### PR DESCRIPTION
**Pull request by agusbusta**
OHLC Data Consolidation and Interval Handling
This update addresses the issue of inconsistent data intervals in OHLC (Open, High, Low, Close) responses. Previously, the API was returning data in 4-hour intervals regardless of the requested interval. The fix implements a data consolidation function to ensure that the returned data matches the requested interval (daily, weekly, etc.).
Key Changes:
Implemented a new consolidate_ohlc_data function:
Processes raw OHLC data and consolidates it based on the specified interval.
Supports daily ('1d') and weekly ('1w') intervals, with extensibility for other intervals.
Ensures that each data point represents the correct time period with accurate OHLC values.
Modified the get_ohlc_coingecko_data function:
Now applies the consolidate_ohlc_data function to the raw API response.
Ensures that the data is properly consolidated before being returned to the client.
Updated data processing flow:
Raw data is fetched from the CoinGecko API.
Data is then consolidated based on the requested interval.
Consolidated data is processed to convert timestamps to ISO format.

### CODE REVIEW

File: routes/chart/chart_olhc.py [hunk 1]
```diff
@@ -19,7 +21,7 @@
 COINGECKO_API_KEY = os.getenv('COINGECKO_API_KEY')
 COINGECKO_API_URL = "https://pro-api.coingecko.com/api/v3"
 BINANCE_API_URL = "https://api3.binance.com/api/v3"
-headers = {'X-Cg-Pro-Api-Key': COINGECKO_API_KEY}
+headers = {'X-Cg-Pro-Api-Key': "CG-xXCJJaHa7QmvQNWyNheKmSfG"}
```
Comment (daviddflix) [inline routes/chart/chart_olhc.py:24]: - This must not happen again, the API key is there, please remove it.

File: routes/chart/chart_olhc.py [hunk 2]
```diff
@@ -19,7 +21,7 @@
...
 
 
 # List of cryptocurrencies we have available
```
Comment (daviddflix) [inline routes/chart/chart_olhc.py:27]: - Use the database to get all available coins.

File: routes/chart/chart_olhc.py [hunk 3]
```diff
@@ -6,6 +7,7 @@
 import requests
 import logging
```
Comment (daviddflix) [inline routes/chart/chart_olhc.py:8]: - Remove the logger, as this is just for debugging.

**Review by daviddflix — CHANGES_REQUESTED** [review]
## Feedback

- Please make sure all endpoints work, and remove any code that we don't use anymore.

**Comment by agusbusta** [comment]
Removed api key form code, now located in env file

**Comment by agusbusta** [comment]
hey @daviddflix its ready to review!

**Review by daviddflix — APPROVED** [review]

**Pull request merged**